### PR TITLE
Support `<Text />` and `<Paragraph />` via the new theme

### PR DESCRIPTION
**Pull request by akleiner2**
**Overview**
Title - backfitting a bunch of the new typography tokens into `<Text />`, `<Paragraph />`, etc.

**Screenshots (if applicable)**

**Documentation**

- [ ] Updated Typescript types and/or component PropTypes
- [ ] Added / modified component docs
- [ ] Added / modified Storybook stories

### CODE REVIEW

File: src/typography/src/Text.js [hunk 1]
```diff
@@ -1,19 +1,15 @@
...
+import useStyleConfig from '../../hooks/use-style-config'
+
+const emptyObject = {}
 
 const Text = memo(
   forwardRef(function Text(props, ref) {
-    const {
-      className,
-      color = 'default',
-      fontFamily = 'ui',
```
Comment (mshwery) [inline src/typography/src/Text.js:11]: We probably need to support `fontFamily` still?

File: src/typography/src/Text.js [hunk 2]
```diff
@@ -1,19 +1,15 @@
...
-import useTextStyle from '../../theme/src/hooks/useTextStyle'
+import useStyleConfig from '../../hooks/use-style-config'
+
+const emptyObject = {}
 
 const Text = memo(
   forwardRef(function Text(props, ref) {
-    const {
-      className,
-      color = 'default',
```
Comment (mshwery) [inline src/typography/src/Text.js:10]: We probably want to keep supporting `color` values from the theme?

File: src/typography/src/Paragraph.js [hunk 3]
```diff
@@ -1,30 +1,22 @@
...
   forwardRef(function Paragraph(props, ref) {
-    const { color = 'default', marginTop, size = 400, ...restProps } = props
+    const { marginTop, size = 400, ...restProps } = props
 
-    const { marginTop: defaultMarginTop, ...textStyle } = useParagraphStyle(
-      size,
-      color
+    const textStyle = useStyleConfig(
+      'Paragraph',
+      { size },
```
Comment (mshwery) [inline src/typography/src/Paragraph.js:14]: Same here – think we need to support theme lookup for `color` and `fontFamily` like we used to do in v5. (We may need to check that inputs still support this as well!)

**Review by mshwery — APPROVED** [review]
seems legit! a couple breaking changes in here w/r/t `color` and `fontFamily` but otherwise looks great!

File: src/typography/src/Text.js (hunk 1 above)
Comment (akleiner2) [reply src/typography/src/Text.js:11]: These happen in the theme itself!

Comment (mshwery) [reply src/typography/src/Text.js:11]: We support these as props though – you can do `<Text fontFamily="mono" />` (see `Code` for instance).

**Pull request merged**